### PR TITLE
Bug Fix: APIUUID is not published to Analytics when Internal-Key is used.

### LINKED ISSUE #2114: Incorrect API-UUID is published to analytics for Internal-Key invocation
### Description:
The API UUID published to the analytics for internal Key invocation, is incorrect at the moment. Instead of the UUID assigned from APIM Publisher, it publishes a hash value.

This is due to authentication Context's API  UUID is not being populated properly when InternalKey Authentication is used.

### Environment details (with versions):
- OS:
- Client:
- Env (Docker/K8s):

---

### PR DESCRIPTION
**Pull request by VirajSalaka**
### Purpose
Fix $subject by populating API UUID to authentication Context when InternalKey Authentication handler and Unsecured Authentication handler is involved in the request flow.

### Issues

Fixes #2114 

### Automation tests
 - Unit tests added: No
 - Integration tests added: No

### Tested environments

Not Tested

---
#### Maintainers: Check before merge
- [x] Assigned 'Type' label
- [x] Assigned the project
- [x] Validated respective github issues
- [x] Assigned milestone to the github issue(s)

### CODE REVIEW

**Comment by VirajSalaka** [comment]
Fixes https://github.com/wso2-enterprise/choreo/issues/5958

File: enforcer/src/main/java/org/wso2/choreo/connect/enforcer/util/FilterUtils.java [hunk 1]
```diff
@@ -202,7 +202,10 @@ public static AuthenticationContext generateAuthenticationContext(RequestContext
         authContext.setStopOnQuotaReach(true);
         authContext.setConsumerKey(null);
         authContext.setCallerToken(null);
-
+        String apiUUID = requestContext.getMatchedAPI().getAPIConfig().getUuid();
+        if (!StringUtils.isEmpty(apiUUID)) {
+            authContext.setApiUUID(apiUUID);
```
Comment (Rajith90) [inline enforcer/src/main/java/org/wso2/choreo/connect/enforcer/util/FilterUtils.java:207]: Can this UUID be empty if there is an actual matched API is found

File: enforcer/src/main/java/org/wso2/choreo/connect/enforcer/util/FilterUtils.java [hunk 2]
```diff
@@ -305,14 +309,16 @@ public static AuthenticationContext generateAuthenticationContext(String tokenId
...
         }
-
         authContext.setApiTier(apiLevelPolicy);
         if (api != null) {
             authContext.setTier(APIConstants.UNLIMITED_TIER);
             authContext.setApiName(api.getAsString(APIConstants.JwtTokenConstants.API_NAME));
             authContext.setApiPublisher(api.getAsString(APIConstants.JwtTokenConstants.API_PUBLISHER));
 
         }
+        if (!StringUtils.isEmpty(apiUUID)) {
```
Comment (Rajith90) [inline enforcer/src/main/java/org/wso2/choreo/connect/enforcer/util/FilterUtils.java:319]: Can this UUID be empty if there is an actual matched API is found

Comment (VirajSalaka) [reply enforcer/src/main/java/org/wso2/choreo/connect/enforcer/util/FilterUtils.java:319]: When the apictl is used, uuid becomes empty

**Review by Rajith90 — APPROVED** [review]

**Pull request merged**